src/main/java/com/minecolonies/entity/ai/citizen/guard/AbstractEntityAIGuard.java

```diff
@@ -100,12 +160,40 @@ protected AbstractEntityAIGuard(@NotNull final JobGuard job)
     {
         super(job);
         super.registerTargets(
-          new AITarget(IDLE, () -> START_WORKING),
-          new AITarget(START_WORKING, () -> GUARD_RESTOCK)
+                new AITarget(this::checkIfExecute, this::getState),
+                new AITarget(IDLE, () -> START_WORKING),
+                new AITarget(START_WORKING, () -> GUARD_RESTOCK),
+                new AITarget(GUARD_GATHERING, this::gathering)
         );
     }
 
     /**
+     * Checks if the worker is in a state where he can execute well.
+     * @return true if so.
+     */
+    private boolean checkIfExecute()
+    {
+        AbstractBuilding building = getOwnBuilding();
+        if(building == null || !(building instanceof BuildingGuardTower))
+        {
+            return true;
+        }
+
+        if(!((BuildingGuardTower) building).shallRetrieveOnLowHealth())
+        {
+            return false;
+        }
+
+        if(worker.getHealth() > 2)
+        {
+            return false;
+        }
+
+        worker.isWorkerAtSiteWithMove(building.getLocation(), 3);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Assign this magic number 3 to a well-named constant, and use the constant instead. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS109)